### ISSUE TITLE
Can't compile without bdb even if --disable-wallet is used.

### ISSUE THREAD

**Issue by jhoenicke** [gandrewstone/bitcoin#1416]
Berkeley DB was always optional if you compile with `configure --disable-wallet`. However, the latest release gives compilation errors when compiled without bdb installed:

```
  CXX      libbitcoin_util_a-allowed_args.o
In file included from ./wallet/walletdb.h:16:0,
                 from wallet/wallet.h:21,
                 from allowed_args.cpp:29:
./wallet/db.h:21:20: fatal error: db_cxx.h: Datei oder Verzeichnis nicht gefunden   
compilation terminated.
```

Possible fix is adding enough `#ifdef ENABLE_WALLET` to `alllowed_args.cpp`.

**Comment by sickpig** [comment]
Will look into it, thanks for the notice.

**Comment by ptschip** [comment]
fixed in PR 1427

if you can give a try , would be appreciated.

@sickpig  any way we can setup travis for our NO wallet build to not
have bdb installed when it compiles?  would catch these problems in the
future...

**Comment by sickpig** [comment]
@ptschip agreed

**Comment by infertux** [comment]
I can confirm this bug was introduced in 1.5.0.0 as my build started to fail: https://travis-ci.org/infertux/chef-bitcoin/builds/442219387 ([raw log](https://api.travis-ci.org/v3/job/442219394/log.txt)).

> any way we can setup travis for our NO wallet build to not have bdb installed when it compiles

Looks like you could remove `libdb4.8*-dev` from the [install block](https://github.com/BitcoinUnlimited/BitcoinUnlimited/blob/release/.travis.yml#L98) and only install it when required.

**Comment by sickpig** [comment]
@jhoenicke, @infertux 
fixed by #1427
